src/App.tsx

```diff
@@ -1,61 +1,114 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { UserList } from './components/UserList/UserList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const getNewId = (allTodos: Todo[]) => {
+  const todoId = allTodos.map(todo => todo.id);
+
+  return Math.max(...todoId) + 1;
+};
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [newTodos, addNewTodos] = useState(todos);
+  const [isError, setError] = useState(false);
+  const [userId, setUserId] = useState(0);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (title && userId) {
+      const todo = {
+        id: getNewId(newTodos),
+        title,
+        userId,
+        completed: false,
+        user: getUser(userId),
+      };
+
+      addNewTodos([...newTodos, todo]);
+      setUserId(0);
+      setTitle('');
+      setError(false);
+    } else {
+      setError(true);
+    }
+  };
+
+  const addTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Consider using a more descriptive name for the function, like `handleTitleChange` instead of `addTitle`. 🤔